### ISSUE TITLE
MDBOOK002 false positives: incorrectly reports existing files as non-existent in v0.11.6

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#239]
## Bug Description
The MDBOOK002 (internal-link-validation) rule is reporting false positive errors for internal links that actually exist and work correctly. The documentation builds perfectly with `mdbook build` when the lint preprocessor is disabled.

## Environment
- **mdbook-lint version**: v0.11.6
- **mdbook version**: v0.4.40
- **OS**: macOS (Darwin 25.0.0)
- **Test project**: https://github.com/joshrotenberg/redisctl

### 1. Configuration in `book.toml`:
```toml
[book]
authors = ["Josh Rotenberg"]
language = "en"
src = "src"
title = "redisctl Documentation"

[output.html]
default-theme = "light"
preferred-dark-theme = "ayu"
git-repository-url = "https://github.com/joshrotenberg/redisctl"
git-repository-icon = "fa-github"

[preprocessor.lint]
command = "mdbook-lint"
renderer = ["html"]
```

### 2. Configuration in `.mdbook-lint.toml`:
```toml
# Temporarily disable rules with known issues
disabled-rules = [
    "MDBOOK010",  # Math block detection - false positives with $ in shell code blocks
    "MDBOOK005",  # Orphaned files - incorrectly checking outside src directory
    "MDBOOK002",  # Internal link validation - false positives with existing files
]

# Don't fail the build on warnings, but show them for fixing
fail-on-warnings = false
fail-on-errors = false

[rules.MD013]
line_length = 120
```

### 3. Example false positive errors:
```
tutorials/common-recipes.md:140:3: error: MDBOOK002/internal-link-validation: Internal link 'production-databases.md' does not resolve to an existing file
tutorials/common-recipes.md:141:3: error: MDBOOK002/internal-link-validation: Internal link 'cicd.md' does not resolve to an existing file
getting-started/quickstart.md:136:3: error: MDBOOK002/internal-link-validation: Internal link '../cloud/overview.md' does not resolve to an existing file
```

### 4. Directory structure showing files DO exist:
```bash
$ ls -la docs/src/tutorials/
-rw-r--r--  15760 Sep 15 20:10 cicd.md
-rw-r--r--   4138 Sep 15 20:10 common-recipes.md
-rw-r--r--   9938 Sep 15 20:10 production-databases.md

$ ls -la docs/src/cloud/
-rw-r--r--  3682 Sep 15 20:10 api-access.md
-rw-r--r--  2085 Sep 15 20:10 overview.md
# ... other files
```

### 5. The markdown content being flagged:
```markdown
## See Also

- [Managing Production Databases](production-databases.md)
- [CI/CD Integration](cicd.md)
- [Best Practices](../reference/best-practices.md)
- [Async Operations](../features/async-operations.md)
```

## Observed Issues

1. **Same directory links** like `filename.md` are incorrectly flagged as non-existent
2. **Relative links with ./** like `./filename.md` are also flagged incorrectly  
3. **Parent directory links** like `../reference/best-practices.md` are flagged even when the target exists
4. **The `disabled-rules` configuration doesn't prevent MDBOOK002 from running** - even with MDBOOK002 in the disabled-rules list, it still reports errors
5. **mdbook itself has no issues** - without mdbook-lint, `mdbook build` succeeds with zero errors and all links work correctly in the generated HTML

## Test Results

- **Total errors**: 99 (96 are MDBOOK002 false positives, 3 are other legitimate issues)
- **Total warnings**: 323 (these appear to be legitimate style issues)
- **mdbook build without lint**: Success, 0 errors

## Expected Behavior

1. MDBOOK002 should correctly resolve relative links to existing files
2. When MDBOOK002 is in the `disabled-rules` list, it should not run at all
3. Link validation should match mdbook's own link resolution logic

## Workaround

Currently, the only workaround is to disable the lint preprocessor entirely in `book.toml`, which means we lose all the valuable style checking that mdbook-lint provides.

## Additional Context

This issue is preventing us from using mdbook-lint in our CI pipeline. The style warnings (MD013, MD047, etc.) are valuable and we'd like to use them, but the false positive errors from MDBOOK002 are blocking adoption.

Happy to provide more details or test any fixes!
